### PR TITLE
Update map reduce word count to use Cloud Dataproc

### PR DESCRIPTION
**Pull request by lesv**
1. update to latest jar’s
2. add Dataproc Initialization Actions script dp-mr-hb-init.sh
3. update README for Cloud Dataproc.
4. Add gsutil command exec to pom.xml
5. Move code out of Third_party as a googler was listed as an author.
6. Update some license files and README for quickstart

### CODE REVIEW

**Comment by lesv** [comment]
@sduskis PTAL
@hegemonic PTAL - Would appreciate README updates
@dennishuo PTAL - README, cluster.sh, and dp-mr-hb-init.sh comments would be appreciated.

**Comment by lesv** [comment]
@AngusDavis PTAL - README, cluster.sh, and dp-mr-hb-init.sh comments would be appreciated.

Note to Angus & Dennis -- I tried to just add to HADOOP_CLASSPATH at the end of hadoop-env.sh & mapred-env.sh, but I kept nuking a something thats a part of Dataproc.  So I copied the jars and that worked.  Suggestions appreciated.

File: java/wordcount-mapreduce/dp-mr-hb-init.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# 
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.set -x -e
+
+# Values you must set
+
+ROLE=$(/usr/share/google/get_metadata_value attributes/role)
```
Comment (dennishuo) [inline java/wordcount-mapreduce/dp-mr-hb-init.sh]: Switch this to attributes/dataproc-role like dataproc-bucket; we'll be removing the non-"dataproc"-prefixed but dataproc-specific metadata keys in the future.

**Comment by lesv** [comment]
In reviewing it again, I guess I managed to get the HADOOP_CLASSPATH thing working.  I should try to get more sleep. :)

**Comment by lesv** [comment]
@maxluebbe FYI

**Pull request merged**